pkg/pxc/queries/queries.go

```diff
@@ -52,6 +66,100 @@ func New(client client.Client, namespace, secretName, user, host string, port in
 	}, nil
 }
 
+func (p *Database) DisableLogBin() error {
+	_, err := p.db.Exec(`SET @@SESSION.SQL_LOG_BIN = off;`)
+	return errors.Wrap(err, "disable bin logging")
+}
+
+func (p *Database) IsReplica() (bool, error) {
+	rows, err := p.db.Query("show replica status")
+	if err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			return false, nil
+		}
+		return false, errors.Wrap(err, "failed to check if pod is replica")
+	}
+	defer rows.Close()
+	cols, err := rows.Columns()
+	if err != nil {
+		return false, errors.Wrap(err, "get columns")
+	}
+	vals := make([]interface{}, len(cols))
+	for i := range cols {
+		vals[i] = new(sql.RawBytes)
+	}
+	for rows.Next() {
+		err = rows.Scan(vals...)
+		if err != nil {
+			return false, errors.Wrap(err, "scan replication status")
+		}
+	}
+	return true, err
+}
+
+func (p *Database) StopAllReplication() error {
+	_, err := p.db.Exec("stop replica")
+	return errors.Wrap(err, "failed to stop replication")
+}
+
+func (p *Database) AddReplicationSource(name, host string, port, weight int) error {
+	_, err := p.db.Exec("SELECT asynchronous_connection_failover_add_source(?, ?, ?, null, ?)", name, host, port, weight)
+	return errors.Wrap(err, "add replication source")
+}
+
+func (p *Database) ReplicationChannelSources(channelName string) ([]ReplicationChannelSource, error) {
+	rows, err := p.db.Query("select host,port,weight from replication_asynchronous_connection_failover where channel_name=?", channelName)
+	if err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			return nil, ErrNotFound
+		}
+		return nil, errors.Wrap(err, "get replication channels")
+	}
+	defer rows.Close()
+	result := make([]ReplicationChannelSource, 0)
+	for rows.Next() {
+		r := ReplicationChannelSource{}
+		err = rows.Scan(&r.Host, &r.Port, &r.Weight)
+		if err != nil {
+			return nil, errors.Wrap(err, "read replication channel info")
+		}
+		result = append(result, r)
+	}
+	return result, nil
+}
+
+func (p *Database) StopReplication(name string) error {
+	_, err := p.db.Exec("stop replica for channel ?", name)
+	return errors.Wrap(err, "stop replication for channel "+name)
+}
+
+func (p *Database) StartReplication(replicaPass string, src ReplicationChannelSource) error {
+	_, err := p.db.Exec(`
+	change master to
```

Comment:
please use new syntax here - https://dev.mysql.com/doc/refman/8.0/en/change-replication-source-to.html
```suggestion
	CHANGE REPLICATION SOURCE TO
```

Reply:
0c293ac